kore/src/Kore/Internal/Pattern.hs

```diff
@@ -135,6 +136,24 @@ isSimplified sideCondition (splitTerm -> (t, p)) =
     TermLike.isSimplified sideCondition t
     && Condition.isSimplified sideCondition p
 
+-- | Checks whether a pattern has simplified children. A term with
+-- a conjunction at the top is simplified if its subterms are simplified.
+hasSimplifiedChildren
+    :: SideCondition.Representation -> Pattern variable -> Bool
+hasSimplifiedChildren sideCondition patt =
+    let term1 = term patt
```

Comment:
Is it necessary to check the children of the `term` here? I thought it would only be necessary to check the children of the `predicate`.